sdk/docs/pages/bcs/index.mdx

```diff
@@ -0,0 +1,331 @@
+# BCS - Binary Canonical Serialization
+
+This small and lightweight library implements
+[Binary Canonical Serialization (BCS)](https://github.com/zefchain/bcs) in TypescriptScript, making
+BCS available in both Browser and NodeJS environments in a type-safe way.`
+
+## Install
+
+To install, add the [`@mysten/bcs`](https://www.npmjs.com/package/@mysten/bcs) package to your
+project:
+
+```sh npm2yarn
+npm i @mysten/bcs
+```
+
+## Quickstart
+
+```ts
+import { bcs } from '@mysten/bcs';
+
+// define UID as a 32-byte array, then add a transform to/from hex strings
+const UID = bcs.array(32, bcs.u8()).transform({
+	input: (id: string) => fromHex(id),
+	output: (id) => toHex(id),
+});
+
+const Coin = bcs.struct('Coin', {
+	id: UID,
+	value: bcs.u64(),
+});
+
+// deserialization: BCS bytes into Coin
+const bcsBytes = Coin.serialize({
+	id: '0000000000000000000000000000000000000000000000000000000000000001',
+	value: 1000000n,
+}).toBytes();
+
+const coin = Coin.parse(bcsBytes);
+
+// serialization: Object into bytes - an Option with <T = Coin>
+const hex = bcs.optionEnum(Coin).serialize({ some: coin }).toHex();
+
+console.log(hex);
+```
+
+## Description
+
+BCS defines the way the data is serialized, and the serialized results contains no type information.
+To be able to serialize the data and later deserialize it, a schema has to be created (based on the
+built-in primitives, such as `string` or `u64`). There are no type hints in the serialized bytes on
+what they mean, so the schema used for decoding must match the schema used to encode the data.
+
+The `@mysten/bcs` library can be used to define schemas that can serialize and deserialize BCS
+encoded data, and and can infer the correct typescript for the schema from the definitions
+themselves rather than having to define them manually.
+
+## Basic types
+
+bcs supports a number of built in base types that can be combined to create more complex types. The
+following table lists the primitive types available:
+
+| Method                | TS Type      | TS Input Type                | Description                                       |
+| --------------------- | ------------ | ---------------------------- | ------------------------------------------------- |
+| `bool`                | `boolean`    | `boolean`                    | Boolean type (converts to `true` / `false`)       |
+| `u8`, `u16`, `u32`    | `number`     | `number`                     | Unsigned Integer types                            |
+| `u64`, `u128`, `u256` | `string`     | `number \| string \| bigint` | Unsigned Integer types                            |
```

Comment:
Maybe we should add a side note about why we decided that these are string output types?